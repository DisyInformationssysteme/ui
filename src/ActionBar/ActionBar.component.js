import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Action from '../Actions/Action';
import css from './ActionBar.scss';

function ActionBar({ selected, actions, multiSelectActions }) {
	const getActionsToRender = () => (selected > 0 ? multiSelectActions : actions);

	const renderActions = actionsToRender =>
		actionsToRender.map((action, index) => <Action key={index} {...action} />);

	const renderSelectedCount = () =>
		<span className={classNames(css['tc-actionbar-selected-count'], 'tc-actionbar-selected-count')}>
			{selected} selected
		</span>;

	const renderActionBar = () => {
		const { left, right } = getActionsToRender();
		const actionBar = [];
		if (left) {
			actionBar.push(
				<div key={0} className={classNames('navbar-left')}>
					{ selected > 0 ? renderSelectedCount() : null}
					{ renderActions(left) }
				</div>
			);
		}
		if (right) {
			actionBar.push(
				<div key={1} className={classNames(css['navbar-right'], 'navbar-right')}>
					{ selected > 0 && !left ? renderSelectedCount() : null}
					{ renderActions(right) }
				</div>
			);
		}
		return actionBar;
	};


	return (
		<nav className={classNames(css['tc-actionbar-container'], 'tc-actionbar-container', 'nav')}>
			{ renderActionBar() }
		</nav>
	);
}

const actionsShape = {
	left: PropTypes.arrayOf(PropTypes.shape(Action.propTypes)),
	right: PropTypes.arrayOf(PropTypes.shape(Action.propTypes)),
};

ActionBar.propTypes = {
	selected: PropTypes.number,
	actions: PropTypes.shape(actionsShape).isRequired,
	multiSelectActions: PropTypes.shape(actionsShape),
};

export default ActionBar;
