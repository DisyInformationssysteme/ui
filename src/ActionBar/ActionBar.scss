$tc-action-bar-background-color: $concrete !default;

.tc-actionbar-container {
	padding: 0 $padding-small;
	background: $tc-action-bar-background-color;

	.tc-actionbar-selected-count {
		margin-right: $padding-larger;
	}

	button {
		margin: $padding-small $padding-larger $padding-small 0;

		&:last-of-type {
			margin-right: 0;
		}
	}

	.navbar-right {
		margin-right: 0;
	}
}
