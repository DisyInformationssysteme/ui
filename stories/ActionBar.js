import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { ActionBar } from '../src/index';

const primary = {
	label: 'Primary',
	icon: 'fa fa-asterisk',
	bsStyle: 'primary',
	onClick: action('You clicked me'),
};

const actions = {
	left: [
		primary,
		{
			label: 'Secondary1',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
	],
	right: [
		{
			label: 'Secondary2',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
		{
			label: 'Secondary3',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
	],
};
const multi3 = {
	label: 'multi3',
	icon: 'fa fa-asterisk',
	onClick: action('You clicked me'),
};

const multiSelectActions = {
	left: [
		{
			label: 'multi1',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
		{
			label: 'multi2',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
	],
	right: [
		multi3,
		{
			label: 'multi4',
			icon: 'fa fa-asterisk',
			onClick: action('You clicked me'),
		},
	],
};

const basicProps = {
	actions,
	multiSelectActions,
};

storiesOf('Actionbar', module)
	.addWithInfo('default', () => (
		<nav>
			<h1>Actionbar</h1>
			<h2>Definition</h2>
			<p>The actionbar component display a group buttons to let the user dispatch actions</p>
			<h2>Examples</h2>
			<p>No Selected, Layout: Left Space Right</p>
			<ActionBar {...Object.assign({}, basicProps, { selected: 0 })} />

			<p>1 Selected, Layout: Left Space Right</p>
			<ActionBar {...Object.assign({}, basicProps, { selected: 1 })} />

			<p>1 Selected, Layout: Right</p>
			<ActionBar
				selected={1}
				actions={{ left: [primary] }}
				multiSelectActions={{ right: [multi3] }}
			/>

		</nav>
	));
