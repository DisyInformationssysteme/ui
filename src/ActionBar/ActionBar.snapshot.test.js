import React from 'react';
import renderer from 'react-test-renderer';

import ActionBar from './ActionBar.component';

jest.mock('react-dom');

const primaryAction = {
	label: 'Primary',
	icon: 'fa fa-asterisk',
	onClick: jest.fn(),
	bsStyle: 'primary',
};

const secondaryAction = {
	label: 'Secondary',
	icon: 'fa fa-asterisk',
	onClick: jest.fn(),
};

describe('ActionBar', () => {
	it('should render without actions', () => {
		// when
		const wrapper = renderer.create(<ActionBar />).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render no-selected actions, all on left ', () => {
		// given
		const props = {
			selected: 0,
			actions: {
				left: [primaryAction, secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render no-selected actions, some on left, the other on right', () => {
		// given
		const props = {
			selected: 0,
			actions: {
				left: [primaryAction],
				right: [secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render no-selected actions, all on right', () => {
		// given
		const props = {
			selected: 0,
			actions: {
				right: [primaryAction, secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render selected count and multi-selected actions, all on left', () => {
		// given
		const props = {
			selected: 1,
			actions: {},
			multiSelectActions: {
				left: [secondaryAction, secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render selected count and multi-selected actions,' +
		' count and some actions on left, the other on right', () => {
		// given
		const props = {
			selected: 1,
			actions: {},
			multiSelectActions: {
				left: [secondaryAction],
				right: [secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render selected count and multi-selected actions, all on right', () => {
		// given
		const props = {
			selected: 1,
			actions: {},
			multiSelectActions: {
				right: [secondaryAction, secondaryAction],
			},
		};
		// when
		const wrapper = renderer.create(
			<ActionBar {...props} />
		).toJSON();
		// then
		expect(wrapper).toMatchSnapshot();
	});
});
