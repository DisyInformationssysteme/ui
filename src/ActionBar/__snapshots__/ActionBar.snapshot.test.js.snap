exports[`ActionBar should render without actions 1`] = `
<nav
  className="tc-actionbar-container nav" />
`;

exports[`ActionBar should render no-selected actions, all on left  1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-left">
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Primary
      </span>
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;

exports[`ActionBar should render no-selected actions, all on right 1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-right">
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Primary
      </span>
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;

exports[`ActionBar should render no-selected actions, some on left, the other on right 1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-left">
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="navbar-right">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;

exports[`ActionBar should render selected count and multi-selected actions, all on left 1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-left">
    <span
      className="tc-actionbar-selected-count">
      1
       selected
    </span>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;

exports[`ActionBar should render selected count and multi-selected actions, all on right 1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-right">
    <span
      className="tc-actionbar-selected-count">
      1
       selected
    </span>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;

exports[`ActionBar should render selected count and multi-selected actions, count and some actions on left, the other on right 1`] = `
<nav
  className="tc-actionbar-container nav">
  <div
    className="navbar-left">
    <span
      className="tc-actionbar-selected-count">
      1
       selected
    </span>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
  <div
    className="navbar-right">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-asterisk"
        title={null} />
      <span>
        Secondary
      </span>
    </button>
  </div>
</nav>
`;
